From 05182db976a5355911fe0bf65e2bee09d882329c Mon Sep 17 00:00:00 2001
From: Miaoqing Pan <quic_miaoqing@quicinc.com>
Date: Thu, 27 Jun 2024 14:57:46 +0800
Subject: [PATCH] PENDING: networkmanager: allow access tmpfs

Signed-off-by: Miaoqing Pan <quic_miaoqing@quicinc.com>
---
 policy/modules/services/networkmanager.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/services/networkmanager.te b/policy/modules/services/networkmanager.te
index 5ae934638..5c624aa93 100644
--- a/policy/modules/services/networkmanager.te
+++ b/policy/modules/services/networkmanager.te
@@ -67,6 +67,8 @@ allow NetworkManager_t NetworkManager_etc_t:dir list_dir_perms;
 allow NetworkManager_t NetworkManager_etc_t:file read_file_perms;
 allow NetworkManager_t NetworkManager_etc_t:lnk_file read_lnk_file_perms;
 
+init_read_script_tmp_files(NetworkManager_t)
+
 manage_dirs_pattern(NetworkManager_t, NetworkManager_etc_rw_t, NetworkManager_etc_rw_t)
 manage_files_pattern(NetworkManager_t, NetworkManager_etc_rw_t, NetworkManager_etc_rw_t)
 filetrans_pattern(NetworkManager_t, NetworkManager_etc_t, NetworkManager_etc_rw_t, { dir file })
-- 
2.25.1

